# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/webgpu/Device.cpp
# Commit: 843c7d19787b
# Full Hash: 843c7d19787b5beb9caa2fbf71192b174738bb92
# Author: Brad Werth <bwerth@mozilla.com>
# Date: 2023-10-17 03:58:35
# Regressor Bug: 1838693
# File Overlap Count: 1
# Description:
#   Bug 1838693 Part 1: Stub in 'lose the device' and trigger it on device lost errors. r=webgpu-reviewers,nical
#   
#   This creates a WebGPUParent::LoseDevice entry point, which informs the
#   WebGPUChild that the device has been lost. The child side is largely a
#   stub, as a later part will rationalize the handling of the device.lost
# ==============================================================================

diff -r e0dd0b10e8fd -r 843c7d19787b dom/webgpu/Device.cpp
--- a/dom/webgpu/Device.cpp	Mon Oct 16 15:29:21 2023 +0000
+++ b/dom/webgpu/Device.cpp	Mon Oct 16 15:31:59 2023 +0000
@@ -114,6 +114,12 @@
   return mLostPromise;
 }
 
+void Device::ResolveLost(Maybe<dom::GPUDeviceLostReason> aReason,
+                         const nsAString& aMessage) {
+  // TODO(BJW): rationalize this with all the other stuff that deals with
+  // the mLostPromise.
+}
+
 already_AddRefed<Buffer> Device::CreateBuffer(
     const dom::GPUBufferDescriptor& aDesc, ErrorResult& aRv) {
   return Buffer::Create(this, mId, aDesc, aRv);